lib/srv/alpnproxy/local_proxy.go

```diff
@@ -398,6 +398,22 @@ func (l *LocalProxy) CheckDBCerts(dbRoute tlsca.RouteToDatabase) error {
 	return trace.Wrap(CheckCertSubject(cert, dbRoute))
 }
 
+// CheckCertExpiry checks the proxy certificates for expiration.
+func (l *LocalProxy) CheckCertExpiry() error {
```

Comment:
Instead of extracting some stuff between `CheckDBCerts` and `CheckCertExpiry` (which would involve some juggling with `l.certsMu` and returning a `x509.Certificate`, I decided to just add another method by copy & paste.

Those two methods are almost identical, with the difference being that `CheckDBCerts` needs to accept `RouteToDatabase` and call `CheckCertSubject`.

I think that's an acceptable solution for now. Once those methods grow, we can think of refactoring them. Let me know if we should approach this in a different way.